pyro/nn/module.py

```diff
@@ -178,15 +182,23 @@ def __init__(self):
         self.active = 0
         self.cache = {}
         self.used = False
+        if _is_module_local_param_enabled():
+            self.param_state = {"params": {}, "constraints": {}}
 
     def __enter__(self):
+        if not self.active and _is_module_local_param_enabled():
+            self._param_ctx = pyro.get_param_store().scope(state=self.param_state)
```

Comment:
Persisting `self.param_state` like this (and in `_pyro_set_supermodule` below) seems to be a reasonable solution for the behavior of vanilla `pyro.param` statements. Values of these parameters are now local to the outermost `PyroModule` in a nested `PyroModule` instance.